### PR TITLE
Refactor config files

### PR DESCRIPTION
**Pull request by bbockelm**
This refactor pulls out the various pieces of the configuration and drops them into separate config files.  This should allow us to better "layer" XCache configuration pieces (and, if nothing else, allow the admin to only read the configs they are interested in).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,8 @@ VERSION := 0.9
 SBIN_FILES := src/stashcache
 INSTALL_SBIN_DIR := usr/sbin
 XROOTD_CONFIG := configs/Authfile-auth configs/Authfile-noauth configs/stashcache-robots.txt configs/xrootd-stashcache-cache-server.cfg configs/xrootd-stashcache-origin-server.cfg
+XROOTD_CONFIGD := configs/config.d/40-osg-http.cfg configs/config.d/40-osg-monitoring.cfg configs/config.d/40-osg-xcache.cfg configs/config.d/50-stashcache-authz.cfg configs/config.d/50-stashcache-logging.cfg configs/config.d/10-osg-site-local.cfg
```
Comment (brianhlin) [inline Makefile]: Can we put each file onto separate lines? It'll make diffs clearer in the future

File: configs/config.d/10-osg-site-local.cfg [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# They have been purposely commented out to cause the server to
+# fail to start unless you edit them.
+
+# A local directory where the daemon can write its cache files.
+# - Must be writable by the xrootd user.
+# - OSG recommends this be at least 1TB.
+# set cachedir = /stash
+
+# The server name to use for monitoring and reporting; this should
+# match the registered resource name in topology.opensciencegrid.org.
```
[the remark this answers is not in the document]
Comment (bbockelm) [reply configs/config.d/10-osg-site-local.cfg]: Done.

File: Makefile (hunk 1 above)
Comment (bbockelm) [reply Makefile]: Done.  Also nuked a few nearby overly-long lines.

**Comment by bbockelm** [comment]
Ok, I'll be merging this.  It's likely under-tested, but I'd like to get this into the hands of @zvada so we can do some final integration.

**Comment by matyasselmeci** [comment]
I don't see any changes to the origin server configs. Do those not need refactoring?

**Comment by bbockelm** [comment]
Those will eventually need refactoring.  Let's do that as a separate ticket, however.

**Comment by zvada** [comment]
> Those will eventually need refactoring. Let's do that as a separate ticket, however.

@bbockelm, @matyasselmeci: was there any ticket created in this regards? I'm about to test [SOFTWARE-3491](https://opensciencegrid.atlassian.net/browse/SOFTWARE-3491) so I wonder whether we addressed somewhere also origin server configs or we still need new ticket for this?

**Comment by matyasselmeci** [comment]
I'll try to find the JIRA ticket (if there is one) but we reworked the origin server configs as well.

**Pull request merged**